### PR TITLE
[Refactor] Extract localization common functionality

### PR DESCRIPTION
**Pull request by emlynmac**
## Purpose
Rework localization code for reuse

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

### CODE REVIEW

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/Style/LocalizationProvider.swift [hunk 1]
```diff
@@ -6,125 +6,15 @@
 import Foundation
 import SwiftUI
 
-protocol LocalizationProviderProtocol {
+protocol LocalizationProviderProtocol: BaseLocalizationProviderProtocol {
     var isRightToLeft: Bool { get }
     func apply(localeConfig: LocalizationOptions)
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/Style/LocalizationProvider.swift:11]: It looks like LocalizationOptions can't be found for tests build

Comment (emlynmac) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/Style/LocalizationProvider.swift:11]: Yup, just fixing that up...
The check is on the UICommon module being able to build

**Comment by palatter** [comment]
Looks good.

**Review by palatter — DISMISSED** [review]

**Review by jimchou-dev — DISMISSED** [review]
LGTM

**Review by vhuseinova-msft — DISMISSED** [review]

**Review by JoshuaLai — APPROVED** [review]

**Review by vhuseinova-msft — APPROVED** [review]

**Pull request merged**